frontend/packages/dev-console/src/components/topology/components/nodes/WorkloadNode.tsx

```diff
@@ -40,18 +37,21 @@ export type WorkloadNodeProps = {
   WithDragNodeProps &
   WithDndDropProps &
   WithContextMenuProps &
-  WithCreateConnectorProps &
-  StateProps;
+  WithCreateConnectorProps;
 
 const ObservedWorkloadNode: React.FC<WorkloadNodeProps> = ({
   element,
   urlAnchorRef,
   canDrop,
   dropTarget,
   dropTooltip,
-  cheURL,
   ...rest
 }) => {
+  const [consoleLinks] = useK8sWatchResource<K8sResourceKind[]>({
+    isList: true,
+    kind: referenceForModel(ConsoleLinkModel),
+  });
+  const cheURL = getCheURL(consoleLinks);
```

Comment:
You should be able to watch the one link with name `che`. But again I think we should avoid a separate watch for each node.

https://github.com/spadgett/console/blob/e47665c50e6f5ea842a2de42e88e60a3c2090452/frontend/packages/dev-console/src/components/topology/topology-utils.ts#L36-L37